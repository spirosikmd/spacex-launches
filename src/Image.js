import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  imageWrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: '400px',
    [theme.breakpoints.down('xs')]: {
      width: '300px',
    },
  },
});

export const Image = ({ classes, alt, src }) => {
  return (
    <div className={classes.imageWrapper}>
      <img src={src} alt={alt} className={classes.image} />
    </div>
  );
};

Image.propTypes = {
  classes: PropTypes.object.isRequired,
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
};

export default withStyles(styles)(React.memo(Image));
