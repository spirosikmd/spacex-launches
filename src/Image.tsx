import React from 'react';
import PropTypes from 'prop-types';
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
} from '@material-ui/core/styles';

const styles = (theme: Theme) =>
  createStyles({
    imageWrapper: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    image: {
      width: '400px',
      [theme.breakpoints.down('xs')]: {
        width: '300px',
      },
    },
  });

interface ImageProps extends WithStyles<typeof styles> {
  alt: string;
  src: string;
}

const Image = ({ classes, alt, src }: ImageProps) => {
  return (
    <div className={classes.imageWrapper}>
      <img src={src} alt={alt} className={classes.image} />
    </div>
  );
};

Image.propTypes = {
  classes: PropTypes.object.isRequired,
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
};

export default withStyles(styles)(React.memo(Image));
