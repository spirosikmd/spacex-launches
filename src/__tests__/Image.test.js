import { Image } from '../Image';

describe('Image', () => {
  let props;

  beforeEach(() => {
    props = {
      src: 'src',
      alt: 'alt',
      classes: {
        imageWrapper: 'imageWrapper',
        image: 'image',
      },
    };
  });

  it('renders', () => {
    expect(shallow(Image, props)).toMatchSnapshot();
  });
});
