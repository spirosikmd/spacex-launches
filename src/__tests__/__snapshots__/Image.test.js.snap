// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders 1`] = `
<WithStyles(Image)
  alt="alt"
  classes={
    Object {
      "image": "image",
      "imageWrapper": "imageWrapper",
    }
  }
  src="src"
>
  <div
    className="Image-imageWrapper-1 imageWrapper"
  >
    <img
      alt="alt"
      className="Image-image-2 image"
      src="src"
    />
  </div>
</WithStyles(Image)>
`;
